lib/catalog/owid/catalog/catalogs.py

```diff
@@ -716,6 +719,80 @@ def find_latest(
     return REMOTE_CATALOG.find_latest(table=table, namespace=namespace, dataset=dataset, version=version)
 
 
+def find_by_indicator(query: str, limit: int = 10) -> CatalogFrame:
+    """Search for tables by indicator name using semantic search.
+
+    Uses the OWID search API to find indicators matching a natural
+    language query, then returns a CatalogFrame that can load the
+    full tables containing those indicators.
+
+    Args:
+        query: Natural language search query (e.g., "solar power generation").
+        limit: Maximum number of results to return (default: 10).
+
+    Returns:
+        CatalogFrame with columns: indicator_title, indicator, plus standard
+        catalog columns (table, namespace, version, dataset, path, etc.).
+
+    Example:
+        ```python
+        from owid.catalog import find_by_indicator
+
+        # Search for indicators
+        results = find_by_indicator("solar power")
+        print(results[["indicator_title", "indicator", "dataset"]])
+
+        # Load the table containing the top result
+        table = results.iloc[0].load()
+        ```
+    """
+    resp = requests.get(
+        OWID_SEARCH_API,
+        params={"query": query, "limit": limit},
+    )
+    resp.raise_for_status()
+    data = resp.json()
+
+    results = data.get("results", [])
+
+    # Build rows for CatalogFrame
+    rows = []
+    for r in results:
+        catalog_path = r.get("catalog_path", "")
+
+        # Parse catalog_path: "grapher/namespace/version/dataset/table#column"
+        # Example: "grapher/unep/2023-12-12/renewable_energy/investments#solar"
+        path_part, _, indicator = catalog_path.partition("#")
+        parts = path_part.split("/")
+
+        if len(parts) >= 4:
+            channel, namespace, version, dataset = parts[0], parts[1], parts[2], parts[3]
+            table = parts[4] if len(parts) > 4 else dataset
+        else:
+            # Fallback if parsing fails
+            channel, namespace, version, dataset, table = "", "", "", "", ""
+
+        rows.append(
+            {
+                "indicator_title": r.get("title"),
+                "indicator": indicator,
+                "score": r.get("score"),
+                "table": table,
+                "namespace": namespace,
+                "version": version,
+                "dataset": dataset,
+                "channel": channel,
+                "path": path_part,
+                "format": "parquet",
+                "is_public": True,
+            }
+        )
```

Comment:
Could we align the ordering of the tables with what the other `find` methods return?

At the moment, `find` method returns (in order):

- 'table'
- 'dataset'
- 'version'
- 'namespace'
- 'channel'
- 'is_public'
- 'dimensions'
- 'path'
- 'formats'

And `find_by_indicator` returns first specific columns:

- 'indicator_title'
- 'indicator'
- 'score'

and then:

- 'table'
- 'namespace'
- 'version'
- 'dataset'
- 'channel'
- 'path'
- 'format'
- 'is_public'

This latter part has a very different order, and also is missing the `dimensions` column. I think we should have both frames consistent (besides the first thre column).